build/postbuild.js

```diff
@@ -0,0 +1,4 @@
+const fs = require('fs');
+const path = require('path');
+
+fs.chmodSync(path.join(__dirname, '../dist', 'solc.js'), '755');
```

Comment:
Why is the `+x` permission missing actually? Is this on Windows? The file in the repo does have that permission and it should be executable in your working copy unless something in the build process is overwriting it.

Reply:
I'm running on Linux but the Typescript output is missing the permission. Breaks the tests. If you have a moment you should be able to replicate the issue.

1. empty this file.
2. run the tests
3. tests error due to permission denied

Reply:
Ah, I see it now. Sorry. So it's actually automatically generated.

> This includes providing an extension type for the binary file solcjs which is now solc.ts which will later be transpiled into solc.js. 

I'm going to review it properly later and now I only wanted to take a quick look through the code so I haven't read all your comments yet :)